### PR TITLE
Get model class from config instead of "sniffing" dataset dimensions

### LINKED ISSUE #664: Replace"dimension sniffing" with configuration to select model grid class in `open_dataset()`
**Priority (low, medium, high)**
Priority: medium

**Is your feature request related to a problem? Please describe.**
This is more about improving the clarity of the design intent of the code, and code readability/understandability than it is problem per se.

**Describe the solution you'd like**
The model grid class to be used for a dataset should be identified in `oceannavigator/datasetconfig.json` so that it is available on `DatasetConfig`. Then `open_dataset()` can use that to determine which model grid class to return.

I think there is also an argument to be made for some renaming of classes to make the design intent and architecture clearer (though, of course, "naming things is hard"). I suggest:
* `Model` becomes `ModelGrid`
* `Mercator` becomes `LatLonModel`, `LatLonGrid`, or `LatLonModelGrid`
* `Nemo` becomes `XYModel`, `IndexGrid`, or `IndexModelGrid`
* `Fvcom` becomes `UnstructuredModel`, `UnstructuredGrid`, or `UnstructuredModelGrid`, but all of those feel too long to me :-(

**Additional context**

re: https://github.com/DFO-Ocean-Navigator/Ocean-Data-Map-Project/blob/1cf39c31ed787216626dd06c98829a271e4dcb66/data/__init__.py#L77

### PR DESCRIPTION
**Pull request by douglatornell**
## Background

Resolves #664 

This PR addresses issue #664. It changes the way we decide what model data class to instantiate in `data.__init__()`. Instead of opening a dataset to inspect its dimensions and decided from them what model data class to use, we read the model data class name from `datasetconfig.json`.

## Why did you take this approach?
* Inspecting the dimensions required us to open the dataset before we really opened the dataset  - a bit convoluted...
* It was not robust with respect to datasets from models that hadn't yet been coded for in `data.__init__()`; e.g. ROMS, perhaps.
* It lead to the "concealed" error in issue #811.

## Anything in particular that should be highlighted?
* Making the model data class an explicit attribute of each dataset in `datasetconfig.json` means that the impact of adding a dataset that requires a new model class will be immediately obvious.
* The change leads to simplification of code in several places, in particular in the `NetCDFData.dimensions` property that was the cause of issue #811.
* We are also able to do away with the special case `meta_only` arg and model data class attribute.

Changes to `datasetconfig.json` in the [DFO-Ocean-Navigator/configurations](https://github.com/DFO-Ocean-Navigator/configurations) repo are required in tandem with this PR; please see configurations repo [PR #17](https://github.com/DFO-Ocean-Navigator/configurations/pull/17).

## Screenshot(s)
N/A

## Checks
- [x] I ran unit tests.
- [x] I've tested the relevant changes from a user POV.

_Hint_ To run all python unit tests run the following command from the root repository directory:
```sh
bash run_python_tests.sh
```

### CODE REVIEW

File: routes/api_v1_0.py
Comment (htmlboss) [inline routes/api_v1_0.py:781]: Nice

**Review by htmlboss — COMMENTED** [review]
Overall looks good. Def try this on a test container before merge given the changes to some critical paths.

Unrelated note I think we should look at the strategy pattern for figuring out how to open a dataset, given that now we have three possible ways: sqlite, zarr, and dap.

File: routes/api_v1_0.py
Comment (douglatornell) [reply routes/api_v1_0.py:781]: Thanks, @htmlboss. I've been running this code in my dev container for the past couple of weeks. So, I'm confident enough in it to move it to staging for some other folks to exercise.

I agree that it would be nice to have a more uniform interface for opening sqlite, zarr, and dap. Will think on it...

**Review by htmlboss — APPROVED** [review]

**Comment by douglatornell** [comment]
@dwayne-hart Nabil has approved this, so when you have time I think it is ready for merge and deployment to staging so that it gets some exercise.

**Pull request merged**
